docs/2019-12-16-Question-Mark-Variables-With-Ensures.md

```diff
@@ -0,0 +1,313 @@
+# Meaning of `?` variables in K
+
+## Summary
+
+The frontend syntax uses question mark variables of the form `?X` to refer to
+existential variables, and uses `ensures` to specify logical constraints on
+those variables.
+These variables are only allowed to appear in the RHS of a rule.
+
+When K rules are translated to Kore axioms, the `?` variables are existentially
+quantified at the top of the RHS and the `ensures` clause is and-ed to the RHS.
+In addition, the regular non-`?` variables are universally quantified at the top
+of the entire rule.
+When both `?`-variables and regular variables are present, we quantify regular
+variables first, followed by `?`-variables.
```

Comment:
Isn't this sentence implied by these?

> the `?` variables are existentially quantified at the top of the RHS

> the regular non-`?` variables are universally quantified at the top of the entire rule

Is this saying something else?

Reply:
I think Xiaohong meant for me to replace the fact that ? vars are quantified at the top of the RHS with them being quantified at the top, as it covers the fresh example, too. I'll reword.